src/Servers/Kestrel/Core/src/Internal/Http3/Http3ControlStream.cs

```diff
@@ -245,6 +254,48 @@ private async ValueTask<long> TryReadStreamHeaderAsync()
         }
     }
 
+    private async ValueTask<long> TryReadWebTransportStreamSessionIdAsync()
+    {
+        // https://ietf-wg-webtrans.github.io/draft-ietf-webtrans-http3/draft-ietf-webtrans-http3.html#section-4.1-1
+        while (_isClosed == 0)
```

Comment:
`_isClosed` is declared as `volatile`. Writes to it are done via `Interlocked.Exchange`, so I'd prefer if the reads are done via `Volatile.Read` (explicitly).

It makes the code also easier to read / follow, as it's clear the memory-ordering is considered.

The same on other places where `_isClosed` is read.

Reply:
Thanks for your review, that's a genuine concern and I noticed that. However, the current implementation([Http3ControlStream](https://github.com/dotnet/aspnetcore/blob/28739c8e10f562c0912b9a3961592c3f1532522d/src/Servers/Kestrel/Core/src/Internal/Http3/Http3ControlStream.cs#L151)) doesn't use the `Volatile.Read` method as you said, and I didn't want to alter any implementation conventions to avoid submitting a PR with numerous changes. I'm happy to address that though.